anonlink/solving/_multiparty_solving.pyx

```diff
@@ -6,27 +7,62 @@ from anonlink.solving._multiparty_solving_inner cimport (
-    merge their groups iff (a) every pair of records is permitted to be
-    matched, and (b) the similarity of every pair of records is above s.
-    The latter requirement is only for tie-breaking: we prefer groups
-    whose minimum pairwise similarity is highest. Any group merge
-    rejected due to requirement (b) will be revisited later with that
-    requirement relaxed.
+    merge their groups iff (a) the proportion of pairs that are
+    permitted to be matched in the groups' records is at least
+    `merge_threshold`, and (b) the proportion of pairs whose similarity
+    is above s in the groups' records is at least `merge_threshold`. The
+    latter requirement is only for tie-breaking: we prefer groups whose
+    minimum pairwise similarity is highest. Any group merge rejected due
+    to requirement (b) will be revisited later with that requirement
+    relaxed.
 
     :param tuple candidates: Candidates, as returned by
         `find_candidates`.
+    :param float merge_threshold: Proportion of agreement between groups
+        required to merge them. In (0, 1].
+            When considering whether to merge two groups, we compute the
+        groups' Cartesian product. We count the pairs in that product
+        whose similarity is above the threshold. We merge the groups if
+        our count is at least `merge_threshold` of the total number of
+        pairs between the two groups.
+            For example, suppose that we have two groups of records: 
+        {A, B} and {C, D}. Let A-C, A-D and B-C be candidate pairs.
+        Setting `merge_threshold` to any value not greater than .75 will
+        permit the two groups to be merged since they share three
+        candidate pairs out of all four pairs. Even though the
+        similarity of B-D is below the threshold, there is enough
+        evidence in the other pairs to conclude that it must be a match.
+        The `merge_threshold` can be thought of as the inverse of the
+        weight we place on that evidence.
+            Higher values increase precision, but decrease recall.
+            A `merge_threshold` lower than one permits two records to be
+        matched without having a corresponding candidate pair. Set it to
+        1.0 to ensure that two records will only be matched if their
+        similarity is above the similarity threshold.
+    :param bool deduplicated: When True, two records that belong to the
+        same dataset will never be in the same group. Default True.
 
-    :return: An tuple of groups. Each group is an tuple of records. Two
-        records are in the same group iff they represent the same
-        entity. Here, a record is a two-tuple of dataset index and
-        record index.
+    :return: An sequence of groups. Each group is an sequence of
+        records. Two records are in the same group iff they represent
+        the same entity. Here, a record is a two-tuple of dataset index
+        and record index.
```

Comment:
This change is unrelated to the rest of the PR. But the docstring was wrong so it was a good opportunity to fix that.